plugins/org.eclipse.n4js/src/org/eclipse/n4js/xtext/workspace/WorkspaceConfigSnapshot.java

```diff
@@ -76,26 +76,59 @@ public URI getPath() {
 	}
 
 	/**
-	 * Find the project with the given nested location.
+	 * Find the project with the given name.
+	 */
+	public ProjectConfigSnapshot findProjectByName(String name) {
+		return name2Project.get(name);
+	}
+
+	/**
+	 * Finds the project having a path that is a prefix of the given URI.
+	 * <p>
+	 * Note the difference to {@link #findProjectContaining(URI)}!
```

Comment:
Could you state here that it returns true also for files outside of source folders but under this project base dir (instead of letting the reader figuring this out by himself).

IMHO: Since there is no client for this method, I would not introduce it. Also it would be easy to add it inside a sublass.